app/javascript/single-file-upload.js

```diff
@@ -0,0 +1,101 @@
+window.addEventListener("DOMContentLoaded", (event) => {
+  const fileInput = document.querySelector('.govuk-file-upload');
+  const maxFileSize = fileInput.dataset.maxSize;
+  const errorHeading = fileInput.dataset.errorHeading;
+  const errorMessage = fileInput.dataset.sizeErrorMessage;
+  const feedback = $(".single-file-upload__message");
+  const saveButtons = document.querySelectorAll('button[type="submit"]');
+
+  if(fileInput){
```

Comment:
`document.querySelectorAll('.govuk-file-upload')?.forEach` instead makes the branch redundant. Rather just have an early `if (!fileInput) return;` since we can't do anything without it anyway.

Reply:
[intentionally didn't use forEach here as it doesn't work on NodeLists like it does an Arrays and apparently the only way you can use it is unsafe](https://dev.to/jess/how-do-i-use-foreach-on-dom-elements-3m9h)

Reply:
It does work the same, only doesn't work on older browsers we don't support